runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java

```diff
@@ -365,8 +366,19 @@ public Response getErroredStatesByTime(@PathParam("stateMachineName") String sta
         if(fromTimestamp.after(toTimestamp)) {
             return Response.status(Response.Status.BAD_REQUEST).entity("fromTime: " + fromTime + " should be before toTime: " + toTime).build();
         }
-
-        return Response.status(200).entity(statesDAO.findErroredStates(stateMachineName, fromTimestamp, toTimestamp, stateName)).build();
+        List<Status> statuses = new ArrayList<Status>();
+        if ("errored".equals(status)) {
+            statuses = Arrays.asList(Status.errored, Status.cancelled, Status.sidelined);
```

Comment:
accept statuses as list from user itself, so that they can pass whatever they want